department-of-veterans-affairs/abd-vro
# WIP - Offers extended deadline to the helm upgrade command used by our deployment scripts

## What was the problem?

We see relatively frequent occurrence of 'context deadline exceeded' errors during deployments.  While the message that accompanies this problem sounds highly ambiguous, the error is not that ambiguous, and the resolution is to just try again in a moment. The github actions runner that is actively running the deployment could be experiencing resource utilization issues, for example: network or CPU utilization. Or it could be the LHDI cluster that is experiencing issues, or a burst of k8 commands etc. While there isn't much we can do to prevent the root cause from occurring, we can avoid timeout errors by increasing the timeout. [more info here](https://uptrace.dev/blog/context-deadline-exceeded.html)

Associated tickets or Slack threads:
- #?

## How does this fix it?[^1]
By setting the timeout parameter to 10 minutes, we'll alleviate most occurrences of the problem.

## How to test this PR
- deploying to dev will test the helm upgrade command


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
